Source/VisualRecognitionV3/VisualRecognition.swift

```diff
@@ -47,39 +47,47 @@ public class VisualRecognition {
     }
     
     /**
-     If the given data represents an error returned by the Visual Recognition service, then return
-     an NSError with information about the error that occured. Otherwise, return nil.
-
+     If the response or data represents an error returned by the Visual Recognition service,
+     then return NSError with information about the error that occured. Otherwise, return nil.
+     
+     - parameter response: the URL response returned from the service.
      - parameter data: Raw data returned from the service that may represent an error.
      */
-    private func dataToError(data: Data) -> NSError? {
+    private func responseToError(response: HTTPURLResponse?, data: Data?) -> NSError? {
+        
+        // ensure data is not nil
```

Comment:
Another case where we need the initial check for successful response code.  I see you have that in most of the services but you missed a few.